csharp/Theorafile.cs

```diff
@@ -133,14 +137,18 @@ tf_callbacks io
 	}
 
 	[DllImport(nativeLibName, EntryPoint = "tf_fopen", CallingConvention = CallingConvention.Cdecl)]
-	private static extern int INTERNAL_tf_fopen(
-		byte[] fname,
+	private static extern unsafe int INTERNAL_tf_fopen(
+		byte* fname,
 		IntPtr file
 	);
-	public static int tf_fopen(string fname, out IntPtr file)
+	public static unsafe int tf_fopen(string fname, out IntPtr file)
 	{
 		file = AllocTheoraFile();
-		return INTERNAL_tf_fopen(UTF8_ToNative(fname), file);
+
+		byte* utf8Fname = Utf8Encode(fname);
+		int result = INTERNAL_tf_fopen(utf8Fname, file);
```

Comment:
"result" ok. I assumed it was a error code or something (to lazy to look).

Reply:
Eh, it’s fine - this one’s good to go, will merge it at the same time as everything else!